src/compiler/checker.ts

```diff
@@ -424,6 +424,8 @@ namespace ts {
             IntrinsicClassAttributes: "IntrinsicClassAttributes"
         };
 
+        const jsxChildrenPropertyName = "children";

```

Comment:
i would say we should treat this the same way we do with `props` instead of hard coding it.